### PR TITLE
[reconfig] Tolerate redundant committee insert

### PR DESCRIPTION
**Pull request by lxfind**
We could be inserting the same committee several times from different sources, e.g. from state sync and from reconfiguration flow. This should be ok.

### CODE REVIEW

**Review by andll — APPROVED** [review]

File: crates/sui-core/src/epoch/committee_store.rs [hunk 1]
```diff
@@ -54,11 +53,10 @@ impl CommitteeStore {
...
 
     pub fn insert_new_committee(&self, new_committee: &Committee) -> SuiResult {
-        let latest_committee = self.get_latest_committee();
-        fp_ensure!(
-            latest_committee.epoch + 1 == new_committee.epoch,
-            SuiError::from("Unexpected new epoch number")
-        );
+        if let Some(old_committee) = self.get_committee(&new_committee.epoch)? {
+            // If somehow we already have this committee in the store, they must be the same.
+            assert_eq!(&old_committee, new_committee);
```
Comment (sadhansood) [inline crates/sui-core/src/epoch/committee_store.rs:58]: Should we return from the function after this assert?

Comment (lxfind) [reply crates/sui-core/src/epoch/committee_store.rs:58]: It's an assert, which would panic?

Comment (sadhansood) [reply crates/sui-core/src/epoch/committee_store.rs:58]: I mean do we still need the insert below if the existing committee in the db is same as what we are going to insert again.

Comment (lxfind) [reply crates/sui-core/src/epoch/committee_store.rs:58]: When this happens, we probably have a fork. So it probably doesn't matte which committee we keep for this epoch.

Comment (andll) [reply crates/sui-core/src/epoch/committee_store.rs:58]: I think @sadhansood is referring to a fact that if we did have a committee and it does match current value, we could have a separate branch for this case and just skip call to `insert`(which will save some serialization/io).

**Pull request merged**